### ISSUE TITLE
Not working anymore?

### ISSUE THREAD

**Issue by O4karitO** [Merubokkusu/discord-spam-bots#21]
Tried it today on a test server(2 users - admin, who is the "spammer" and a dummy user). 
It gets the user list however then it just stops(no errors), nothing

**Comment by Merubokkusu** [comment]
Which spammer did you use?

**Comment by O4karitO** [comment]
dm text
Actually, just checked, the issue is that server.members is always length = 1 - the spammer himself.
So it doesn't see other users
server.name returns a correct name so it's kina weird

**Comment by Merubokkusu** [comment]
Hmm, whats weird is that I just tried it and it works 🤔 
What OS are you using?

**Comment by O4karitO** [comment]
windows 8.1. 
python 3.6.3

**Comment by O4karitO** [comment]
interesting. just tested it on another group, seems to be working. seems weird.

**Comment by Merubokkusu** [comment]
Server might have a higher moderation setting on


![img](https://a.safe.moe/bBkGO.png)

**Comment by O4karitO** [comment]
Nope, also unrestricted

**Comment by Merubokkusu** [comment]
Wrong server ID maybe?

**Comment by O4karitO** [comment]
triple checked it.

**Comment by Merubokkusu** [comment]
![img](http://media.giphy.com/media/a5viI92PAF89q/giphy.gif)


Honestly I'm not sure, I thought it could have been that it was because there was only 1 user in the server but I recreated the way you said and it worked 🤔

**Comment by O4karitO** [comment]
Interesting, it works in other servers. Another issue I noticed is this error happens from time to time:
dictionary changed size during iteration

**Comment by Merubokkusu** [comment]
Yeah, its a bug I tried to fix but will happen from time to time.

Pretty much someone joins the server as it counting how many users are in it.

**Comment by O4karitO** [comment]
for member in list(UserList):
this seems to fix it

**Comment by Merubokkusu** [comment]
Whoa, yeah you're right that does fix it.

**Comment by O4karitO** [comment]
Are you considering adding proxy support(you already added multi-account support but without proxy it's kinda useless)?

**Comment by Merubokkusu** [comment]
From what I've researched the only way to add proxy support is to edit the discord API itself. 

https://github.com/Rapptz/discord.py/issues/98

**Comment by O4karitO** [comment]
good news for discord, otherwise spammers would be all over the place. Though they are anyway)

**Comment by wallopthecat** [comment]
I'm getting something similar, can't get the userlist for a 50k+ server

**Comment by Merubokkusu** [comment]
@wallopthecat
50k users?
If that's how many users it is the problem is that it would take so long to scrape all of the usernames.

I never expected it to be used in such a high volume, I'll look into a way to make it work better.

**Comment by Merubokkusu** [comment]
@wallopthecat 
https://github.com/Merubokkusu/discord-spam-bots/commit/7ad0b003c63812abc4ef9415a1647d7c09b84a4c Hopefully this fixes it.

**Comment by wallopthecat** [comment]
Thanks for the quick commit. Tested option 4 a new token and new IP and we are golden

**Comment by wallopthecat** [comment]
I take that back. Seems to have a hard time finding users that have not been messaged already, and leaks memory while it loops through the members is has already sent to:
```
xyz was already messaged
```

**Comment by Merubokkusu** [comment]
Not sure about the memory leak, but how long does it go before it has trouble finding users?

**Comment by wallopthecat** [comment]
500 users or so

**Comment by Merubokkusu** [comment]
Hmm, I'm not sure how I could do it without scraping them.

If you were to scrape 50K user IDS the file size would be around 900mb.

How often does it tell you the user was already messaged?
